pkg/registry/pod/etcd/etcd.go

```diff
@@ -280,7 +280,7 @@ func (r *ProxyREST) NewConnectOptions() (runtime.Object, bool, string) {
 }
 
 // Connect returns a handler for the pod proxy
-func (r *ProxyREST) Connect(ctx api.Context, id string, opts runtime.Object) (rest.ConnectHandler, error) {
+func (r *ProxyREST) Connect(ctx api.Context, id string, opts runtime.Object, _ rest.Responder) (http.Handler, error) {
```

Comment:
The responder object needs to be passed to the upgradeAwareProxyHandler so that it sets the error on it, instead of setting it on its own instance var:

https://github.com/smarterclayton/kubernetes/blob/connect_responder/pkg/registry/generic/rest/proxy.go#L77
https://github.com/smarterclayton/kubernetes/blob/connect_responder/pkg/registry/generic/rest/proxy.go#L110
https://github.com/smarterclayton/kubernetes/blob/connect_responder/pkg/registry/generic/rest/proxy.go#L128-L150

The RequestError method on that is also not needed anymore:
https://github.com/smarterclayton/kubernetes/blob/connect_responder/pkg/registry/generic/rest/proxy.go#L63